pkg/security/podsecuritypolicy/provider.go

```diff
@@ -265,7 +265,27 @@ func (s *simpleProvider) ValidatePodSecurityContext(pod *api.Pod, fldPath *field
 			}
 		}
 	}
+	if len(pod.Spec.Volumes) > 0 && len(s.psp.Spec.AllowedFlexVolumes) > 0 && psputil.PSPAllowsFSType(s.psp, extensions.FlexVolume) {
```

Comment:
move inside the loop, parallel to the `if fsType == extensions.HostPath {` block